src/variableMenu.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { instrumentOperationAsVsCodeCommand } from "vscode-extension-telemetry-wrapper";
+
+export function registerVariableMenuCommands(context: vscode.ExtensionContext): void {
+    vscode.workspace.onDidChangeConfiguration((event) => {
+        if (event.affectsConfiguration("java.debug.settings")) {
+            updateContextKeys();
+        }
+    });
+    // Initialize the context keys
+    updateContextKeys();
+
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.showHex", () => turnOnFormat("showHex")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.notShowHex", () => turnOffFormat("showHex")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.showQualifiedNames", () => turnOnFormat("showQualifiedNames")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.notShowQualifiedNames", () => turnOffFormat("showQualifiedNames")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.showStaticVariables", () => turnOnFormat("showStaticVariables")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.notShowStaticVariables", () => turnOffFormat("showStaticVariables")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.showLogicalStructure", () => turnOnFormat("showLogicalStructure")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.notShowLogicalStructure", () => turnOffFormat("showLogicalStructure")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.showToString", () => turnOnFormat("showToString")));
+    context.subscriptions.push(
+        instrumentOperationAsVsCodeCommand("java.debug.variables.notShowToString", () => turnOffFormat("showToString")));
+}
+
+function turnOnFormat(key: string) {
```

Comment:
Merge `turnOnFormat` & `turnOffFormat` to something like `updateFormatSetting(key: string, value: any)`?